### DISCUSSION TITLE
[API Request] Return ISRC with song info

### DISCUSSION THREAD

**Discussion by gabeklavans** [opensubsonic/open-subsonic-api#123] [API requests]
### Type of change

API tweak

### Proposal description

I propose to return the [ISRC](https://musicbrainz.org/doc/ISRC), if present in the tags, as an optional field of a retrieved song, i.e. in the Child object, with key `isrc`. Similar to the MusicBrainz Recording ID being returned with key `musicBrainzId`.

This value is useful when matching tracks across various services, such as [Spotify](https://developer.spotify.com/documentation/web-api/reference/get-track). MusicBrainz Picard attempts to add it to tagged songs, and ISRC is part of [ID3](https://id3.org/id3v2.3.0).

### Backward compatibility impact

This should not impact backwards compatibility, as it only adds a new optional return field.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

Add optional `isrc` string field to the Child object.

Can also be added to the Song object, though I'm not sure if that's used yet.

**Comment by gabeklavans** [comment]
It seems to me it would be relatively straight forward to add this to the spec. If there is no discussion as to why this might not be a good idea, I might draft a PR for it sometime soon.

**Reply by deluan** [reply]
:+1: I think this is straightforward, and should be added.

**Reply by Tolriq** [reply]
While I'm all for new fields, as a reminder when discussing about the codecs there were lot of remarks about size of data sent that clients may not need.
I don't think that's an issue, and won't be an issue with the new endpoints but this should be acknowledged.

**Reply by gabeklavans** [reply]
Yeah that’s a fair point. Although, my opinion is that the solution for that problem would be to implement a way to request which fields are able to be returned for a given request, rather than prevent more useful fields from being added to the spec at all. But yeah, to your other point, that may not be an issue worth solving yet.

**Reply by Tolriq** [reply]
The new endpoints will have field selection to address this so no more or less an issue in the future, for now usefulness or at least the number or servers and clients that will the field should be checked before adding a field that maybe no one would use.

**Reply by gabeklavans** [reply]
> for now usefulness or at least the number or servers and clients that will the field should be checked before adding a field that maybe no one would use

On that note I'm definitely going to open a PR to implement this in [gonic](https://github.com/sentriz/gonic) since I use it as my server and this new field will directly help some of the work I'm doing on spotify-to-subsonic track matching programs. So that's at least one popular server that will (hopefully) use it!

**Reply by gabeklavans** [reply]
> On that note I'm definitely going to open a PR to implement this in [gonic](https://github.com/sentriz/gonic) since I use it as my server and this new field will directly help some of the work I'm doing on spotify-to-subsonic track matching programs. So that's at least one popular server that will (hopefully) use it!

maybe I can quickly get @sentriz's take on this ^ before moving further
